rviz_common/src/rviz_default_plugins/point_cloud_common.hpp

```diff
@@ -0,0 +1,282 @@
+  };
+
+  typedef std::shared_ptr<CloudInfo> CloudInfoPtr;
+  typedef std::deque<CloudInfoPtr> D_CloudInfo;
+  typedef std::vector<CloudInfoPtr> V_CloudInfo;
+  typedef std::queue<CloudInfoPtr> Q_CloudInfo;
+  typedef std::list<CloudInfoPtr> L_CloudInfo;
+
+  explicit PointCloudCommon(rviz_common::Display * display);
+  ~PointCloudCommon();
+
+  void initialize(rviz_common::DisplayContext * context, Ogre::SceneNode * scene_node);
+
+  void fixedFrameChanged();
+  void reset();
+  void update(float wall_dt, float ros_dt);
+
+  void addMessage(const sensor_msgs::msg::PointCloud::ConstSharedPtr & cloud);
+  void addMessage(const sensor_msgs::msg::PointCloud2::ConstSharedPtr & cloud);
+
+//  ros::CallbackQueueInterface* getCallbackQueue() { return &cbqueue_; }
+
+  rviz_common::Display * getDisplay() {return display_;}
+
+  bool auto_size_;
+
+  rviz_common::properties::BoolProperty * selectable_property_;
+  rviz_common::properties::FloatProperty * point_world_size_property_;
+  rviz_common::properties::FloatProperty * point_pixel_size_property_;
+  rviz_common::properties::FloatProperty * alpha_property_;
+  rviz_common::properties::EnumProperty * xyz_transformer_property_;
+  rviz_common::properties::EnumProperty * color_transformer_property_;
+  rviz_common::properties::EnumProperty * style_property_;
+  rviz_common::properties::FloatProperty * decay_time_property_;
+
+  void setAutoSize(bool auto_size);
+
+public Q_SLOTS:
+  void causeRetransform();
+
+private Q_SLOTS:
+  void updateSelectable();
+  void updateStyle();
+  void updateBillboardSize();
+  void updateAlpha();
+  void updateXyzTransformer();
+  void updateColorTransformer();
+  void setXyzTransformerOptions(rviz_common::properties::EnumProperty * prop);
+  void setColorTransformerOptions(rviz_common::properties::EnumProperty * prop);
+
+private:
+  /**
+   * \brief Transforms the cloud into the correct frame, and sets up our renderable cloud
+   */
+  bool transformCloud(const CloudInfoPtr & cloud, bool fully_update_transformers);
+
+  void processMessage(const sensor_msgs::msg::PointCloud2::ConstSharedPtr & cloud);
+  void updateStatus();
+
+  PointCloudTransformerPtr getXYZTransformer(
+    const sensor_msgs::msg::PointCloud2::ConstSharedPtr & cloud);
+  PointCloudTransformerPtr getColorTransformer(
+    const sensor_msgs::msg::PointCloud2::ConstSharedPtr & cloud);
+  void updateTransformers(const sensor_msgs::msg::PointCloud2::ConstSharedPtr & cloud);
+  void retransform();
+  void onTransformerOptions(V_string & ops, uint32_t mask);
+
+  void loadTransformers();
+  void loadTransformer(
+    PointCloudTransformerPtr trans,
+    std::string name,
+    const std::string & lookup_name);
+
+  float getSelectionBoxSize();
+  void setPropertiesHidden(const QList<rviz_common::properties::Property *> & props, bool hide);
+  void fillTransformerOptions(rviz_common::properties::EnumProperty * prop, uint32_t mask);
+
+//  ros::AsyncSpinner spinner_;
+//  ros::CallbackQueue cbqueue_;
```

Comment:
TODO or remove, you can add me as the TODO person and mention something about the executors and callback groups so I can find it again.